gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/channel/TestChannelizer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.server.channel;
+
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import org.apache.tinkerpop.gremlin.server.Channelizer;
+
+public interface TestChannelizer extends Channelizer {
```

Comment:
Have you considered using a Decorator/Wrapper instead of inheritance? This might allow to you to make just a single new class rather than three classes. However, initialization might be clunky because of how they need to be passed to settings.channelizer.

Reply:
I agree that inheritance isn't ideal here, I couldn't find a reasonable way of initializing the channelizer using a decorator. The channelizer being passed into the server as a string of the class name is pretty limiting. I would be open to other ideas if you have any.